adapters/repos/db/repo.go

```diff
@@ -277,7 +277,12 @@ type Config struct {
 }
 
 func (db *DB) GetIndexLike(className schema.ClassName) IndexLike {
-	return db.GetIndex(className)
+	index := db.GetIndex(className)
+	if index == nil {
+		return nil
+	}
+
+	return index
```

Comment:
hmm, can we revert those changes here? or we have to fetch `index` here and do the `nil check`?

Reply:
we have to do check to avoid panics, because the method return an interface and once we return as is it became too late and the deference (*Index), so with this we return concerte  nil

Reply:
but in case of `index` being `nil` we still return `nil`, my point is that this additional check is no different from just doing `return db.GetIndex(className)`

Reply:
there is actually https://github.com/weaviate/weaviate/pull/9022/files#r2307492317